src/libcore/slice.rs

```diff
@@ -666,6 +666,60 @@ macro_rules! iterator {
                 let exact = diff / (if size == 0 {1} else {size});
                 (exact, Some(exact))
             }
+
+            #[inline]
+            fn count(self) -> usize {
```

Comment:
I think that `size_hint` doesn't consume the iterator, so I don't think this consumes the iterator?